### PR TITLE
[I420 color conversion] I420toRGB/I420toBGR reference implementation

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
 - Based on already existing NV12toRGB/BGR implementation
 - Reference implementation of color conversion from I420 to RGB/BGR formats
 - Specification is under separate PR: https://github.com/openvinotoolkit/openvino/pull/8292
 - Tests:
     - ngraph: type_prop and visitor tests
     - Template plugin: reference evaluate
     - IE shared tests: compliance and accuracy tests
     - CPU: compliance tests (currently passed due to fallback)
     - IE functional tests: serialization tests

### TODO:
 - OpenCV compliance tests will be added during integration of new ops to 'OV 2.0 preprocessing', like it is done for NV12

### Tickets:
 - 62175
 - 71211

### CODE REVIEW

File: ngraph/core/src/op/util/convert_color_i420_base.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    auto y_type = get_input_element_type(0);
+    const auto& shape_y = get_input_partial_shape(0);
+    const auto one_channel_nhwc_shape =
+        PartialShape({Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic(), 1});
+    NODE_VALIDATION_CHECK(this,
+                          shape_y.compatible(one_channel_nhwc_shape),
+                          "Y input shall have 4 dimensions (N, H, W, C) with channels dimension equal to 1");
+    auto out_shape = shape_y;
+    auto out_type = y_type;
+    if (out_shape.rank().is_dynamic()) {
```
Comment (ilya-lavrenov) [inline ngraph/core/src/op/util/convert_color_i420_base.cpp:49]: why `shape_y` can have dynamic rank? I suppose it's always `4`?

File: ngraph/core/src/op/util/convert_color_i420_base.cpp [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+bool evaluate_i420_convert(const ov::HostTensorVector& input_values,
+                           const ov::HostTensorPtr& output_value,
+                           bool single_tensor,
+                           ov::op::util::ConvertColorI420Base::ColorConversion conv_format) {
+    bool rc = false;
+    switch (input_values[0]->get_element_type()) {
+        NGRAPH_TYPE_CASE(evaluate_i420_convert, u8, input_values, output_value, single_tensor, conv_format);
+        NGRAPH_TYPE_CASE(evaluate_i420_convert, f16, input_values, output_value, single_tensor, conv_format);
+        NGRAPH_TYPE_CASE(evaluate_i420_convert, bf16, input_values, output_value, single_tensor, conv_format);
```
Comment (ilya-lavrenov) [inline ngraph/core/src/op/util/convert_color_i420_base.cpp:182]: we don't need f16, bf16, f64

File: ngraph/core/src/op/util/convert_color_i420_base.cpp (hunk 1 above)
Comment (nosovmik) [reply ngraph/core/src/op/util/convert_color_i420_base.cpp:49]: During build stage, PartialShape can be fully dynamic and be specified on 'inference' stage. This looks standard ngraph practice allowing fully dynamic shapes and element types on build stage.

**Review by ilya-lavrenov — APPROVED** [review]

File: ngraph/core/src/op/util/convert_color_i420_base.cpp (hunk 2 above)
Comment (nosovmik) [reply ngraph/core/src/op/util/convert_color_i420_base.cpp]: Fixed

File: ngraph/core/src/op/util/convert_color_i420_base.cpp [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                              shape_v);
+        auto shape_uv = shape_u;
+        PartialShape::merge_into(shape_uv, shape_v);
+        if (shape_uv.rank().is_static()) {
+            if (!shape_uv[i420_op::H_DIM].is_dynamic()) {
+                shape_uv[i420_op::H_DIM] *= 2;
+            }
+            if (!shape_uv[i420_op::W_DIM].is_dynamic()) {
+                shape_uv[i420_op::W_DIM] *= 2;
+            }
```
Comment (jane-intel) [inline ngraph/core/src/op/util/convert_color_i420_base.cpp:102]: Basically we can multiply dimensions without checking if it is dynamic. Please [see](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/dimension.cpp#L38) 
```suggestion
            shape_uv[i420_op::H_DIM] *= 2;
            shape_uv[i420_op::W_DIM] *= 2;
```

Feel free to change it in the next PR -- as you stated in the description there will be a follow up PR.

**Review by jane-intel — COMMENTED** [review]
Looks good with few comments. Agree to merge as is

**Review by jane-intel — APPROVED** [review]

File: docs/template_plugin/tests/functional/op_reference/convert_color_i420.cpp [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        const auto in2 = std::make_shared<op::v0::Parameter>(input2.type, input2.shape);
+        const auto in3 = std::make_shared<op::v0::Parameter>(input3.type, input3.shape);
+        std::shared_ptr<Node> conv;
+        conv = std::make_shared<T>(in1, in2, in3);
+        auto res = std::make_shared<op::v0::Result>(conv);
+        return std::make_shared<Function>(ResultVector{res}, ParameterVector {in1, in2, in3});
+    }
+};
+
+TEST_F(ReferenceConvertColorI420LayerTest, CompareWithHardcodedRefs_r_u8_single_rgb) {
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/convert_color_i420.cpp:49]: Looks like you can use parametrized tests

File: ngraph/core/include/openvino/op/util/convert_color_i420_base.hpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+public:
+    OPENVINO_OP("ConvertColorI420Base", "util");
+
+    void validate_and_infer_types() override;
+
+    bool visit_attributes(AttributeVisitor& visitor) override;
+
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    bool evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const override;
```
Comment (ilyachur) [inline ngraph/core/include/openvino/op/util/convert_color_i420_base.hpp:77]: I think we can use new evaluate API

**Review by ilyachur — APPROVED** [review]
I propose to use new evaluate API

File: ngraph/core/include/ngraph/op/i420_to_rgb.hpp [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+//
+
+#pragma once
+
+#include "openvino/op/i420_to_rgb.hpp"
+
+namespace ngraph {
+namespace op {
+namespace v8 {
+using ov::op::v8::I420toRGB;
```
[the remark this answers is not in the document]
Comment (nosovmik) [reply ngraph/core/include/ngraph/op/i420_to_rgb.hpp:12]: @ilyachur Can you comment on this?
For NV12, you've requested to have such aliases in 'ngraph' namespace. See https://github.com/openvinotoolkit/openvino/pull/7601#discussion_r714605602

**Pull request merged**